infra/modules/gcp-tf-runner/main.tf

```diff
@@ -5,41 +5,35 @@
 # times throughout the code base, and includes some hard-coded convention stuff that imho is better
 # to have in one place.
 
+locals {
+  tf_runner_passed_in = var.tf_gcp_principal_email != null && try(length(var.tf_gcp_principal_email) > 0, false)
+}
+
+
 
 # in cloud shell, this seems to return {"email":"", "id":""}
 # in any env, this is NEVER the gcp service account configured via provider block
 # (eg, google.impersonate_service_account = "terraform@...")
+# similarly, data.google_client_config is the same
+# and locally, data.google_service_account_id_token only works when auth'd as a service account
 data "google_client_openid_userinfo" "me" {
-
+  count = local.tf_runner_passed_in ? 0 : 1
 }
 
-#
-
-# if no 'email' field from 'google_client_openid_userinfo', generate id token for the current user
-# and parse email from it.
-# such parsing is explicitly allowed by Google; see https://cloud.google.com/docs/authentication/token-types#id
-#
-# however, this still appears to be the email of the authenticated user, not service account that
-# it's impersonating via the provider configuration
-#
-# we could pass in the service account email as a variable; but that pollutes a LOT of interfaces
-# and calls into question why we even use this module at all
+# this seems to be what Cloud Shell relies on??
 data "google_service_account_id_token" "identity" {
-  count = data.google_client_openid_userinfo.me.email == "" ? 1 : 0
+  count = !local.tf_runner_passed_in && try(data.google_client_openid_userinfo.me[0].email, "") == "" ? 1 : 0
```

Comment:
this works in cloud shell, even though that's not *really* a service account, afiak ... weird. But this is what makes cloud shell work *without* impersonation.